File: android_mobile/src/main/java/com/alexstyl/specialdates/ui/base/MementoPreferenceActivity.java
package com.alexstyl.specialdates.ui.base;

import android.content.Intent;
import android.support.v4.app.Fragment;

import java.util.List;

public class MementoPreferenceActivity extends ThemedMementoActivity {

    @Override
    protected boolean shouldUseHomeAsUp() {
        return true;
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        List<Fragment> fragments = getSupportFragmentManager().getFragments();
        if (fragments != null) {
            for (Fragment frag : fragments) {
                frag.onActivityResult(requestCode, resultCode, data);
            }
        }
    }

}

File: android_mobile/src/main/java/com/alexstyl/specialdates/ui/base/MementoPreferenceActivity.kt
package com.alexstyl.specialdates.ui.base

import android.content.Intent

open class MementoPreferenceActivity : ThemedMementoActivity() {

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent) {
        super.onActivityResult(requestCode, resultCode, data)
        val fragments = supportFragmentManager.fragments
        if (fragments != null) {
            for (frag in fragments) {
                frag.onActivityResult(requestCode, resultCode, data)
            }
        }
    }

}
